src/GraphQL/Execution/ExecutionStrategy.cs

```diff
@@ -154,7 +154,7 @@ protected virtual bool ShouldIncludeNode(ExecutionContext context, IHasDirective
         /// </summary>
         protected virtual void SetSubFieldNodes(ExecutionContext context, ObjectExecutionNode parent)
         {
-            var parentType = parent.GetObjectGraphType(context.Schema)!;
+            var parentType = parent.GetObjectGraphType(context.Schema) ?? parent.GraphType;
```

Comment:
This change makes it so that, for an interface, if `IsTypeOf` returns `false` for all configured object types, the interface graph type is used to resolve the query.  Previously some weird meaningless exception would occur.  This should not break GraphQL spec.

Reply:
> for an interface, if IsTypeOf **returns false for all configured object types**, the interface graph type is used to resolve the query

Is it normal? It seems to me that this situation is a configuration error and when returning the interface we would just hide problem instead helping to solve it.

>  Previously some weird meaningless exception would occur. 

So maybe to track this case here and throw other exception with user-friendly error message?

Reply:
I suppose `GetObjectGraphType` should always return something (not null) with the exception of root node.

Reply:
I do not see why the returned value must be any specific object graph type defined in the schema.  There is no GraphQL spec, as far as I can tell, that requires this.  Although such a configuration may be rare now, it may be more common if users use CLR types to generate their schema, and in so doing, map an interface for which they do not also map each and every CLR class that implements the interface.

Reply:
I compare this to a .NET public interface with one or more private implementations.

Reply:
I think I understand your reasoning. Yes, field resolver returns some "runtime" object and then execution engine use this object against some graph type. For concrete graph type it is this type. For interfaces and unions this type should be somehow inferred. We have `ResolveType`+`IsTypeOf` to solve this task. If you use interface graph type as a "fallback" in case of `null`, then it will work I'm sure, but it is possible will work not at all as the author conceived. Imagine some query with fragment like
```graphql
query {
  user {
    ... on Administrator {
       login
       password
    }
    ... on Guest {
       image
    }
 }
}
```

`Query.user` is interface graph type `Person`. Schema misconfigured somehow (I don't know exactly how) so `GetObjectGraphType` returns `null` instead of `Administrator` (or `Guest`) and then you start using `Person` as resulting type. I think that `CollectFields` would return wrong selection set.

Reply:
I already added one test for this, but I'll add additional tests if you like.

Reply:
I added a whole bunch more tests.  Everything works exactly as expected.

Reply:
I’m not sure that the tests are testing exactly what I'm talking about, but it is not so important to me now. Too narrow case.